subsys/logging/log_core.c

```diff
@@ -46,38 +50,50 @@ static inline void msg_finalize(struct log_msg *msg,
 
 void log_0(const char *str, struct log_msg_ids src_level)
 {
-	struct log_msg *msg = log_msg_create_0(str);
+	if(IS_ENABLED(CONFIG_LOG_FRONTEND)) {
+		log_frontend_nrf_0(str, src_level);
+	} else {
+		struct log_msg *msg = log_msg_create_0(str);
 
-	if (msg == NULL) {
-		return;
+		if (msg == NULL) {
+			return;
+		}
+		msg_finalize(msg, src_level);
 	}
-	msg_finalize(msg, src_level);
 }
 
 void log_1(const char *str,
 	   u32_t arg0,
 	   struct log_msg_ids src_level)
 {
-	struct log_msg *msg = log_msg_create_1(str, arg0);
+	if(IS_ENABLED(CONFIG_LOG_FRONTEND)) {
+		log_frontend_nrf_1(str, arg0, src_level);
+	} else {
```

Comment:
Please check some other files in zephyr.
I  am not sure if `else` shall be in new line.